common/flatpak-dir.c

```diff
@@ -11995,22 +12042,32 @@ sort_strv (char **strv)
 char **
 flatpak_dir_get_default_locale_languages (FlatpakDir *self)
 {
-  char **langs = NULL;
+  g_autoptr(GPtrArray) langs = g_ptr_array_new ();
+  g_autoptr(GDBusProxy) localed_proxy = NULL;
+  g_autoptr(GDBusProxy) accounts_proxy = NULL;
 
   if (flatpak_dir_is_user (self))
     return flatpak_get_current_locale_langs ();
 
-  /* If proxy is not NULL, it means that AccountService exists
-   * and gets the list of languages from AccountService. */
-  g_autoptr(GDBusProxy) proxy = get_accounts_dbus_proxy ();
-  if (proxy != NULL)
-    langs = get_locale_langs_from_accounts_dbus (proxy);
+  /* First get the system default locales */
+  localed_proxy = get_localed_dbus_proxy ();
+  if (localed_proxy != NULL)
+    get_locale_langs_from_localed_dbus (localed_proxy, langs);
 
-  /* If langs is NULL, it means using all languages */
-  if (langs == NULL)
-    langs = g_new0 (char *, 1);
+  /* Now add the user account locales from AccountsService. If accounts_proxy is
+   * not NULL, it means that AccountsService exists */
+  accounts_proxy = get_accounts_dbus_proxy ();
+  if (accounts_proxy != NULL)
+    get_locale_langs_from_accounts_dbus (accounts_proxy, langs);
 
-  return langs;
+  /* If none were found, fall back to using all languages */
+  if (langs->len == 0)
+    return g_new0 (char *, 1);
```

Comment:
Is this early return necessary? If I’m reading it correctly, the code below should do exactly the same thing as this when `langs->len == 0`.

Reply:
Even if you're right, I think the explicit way it's written is better, so people reading the code understand it might return an empty array.